### PR TITLE
NEXT-31267 - Vue3 update guide

### PR DESCRIPTION
**Pull request by seggewiss**

### CODE REVIEW

File: guides/updates/administration/vue3.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+  title: Vue 3 update
+  position: 10
+
+---
+
+# Vue 3 update
+
+## Introduction
+
+The Shopware administration uses Vue.js `2`, which will reach its end of life (EOL) **on December 31st 2023**. To deliver up-to-date and maintainable software, the administration will use Vue.js `3` from Shopware version `6.6` and upwards. If you are unfamiliar with the changes from Vue.js `2` to Vue.js `3`, please refer to this (official guide)[https://v3-migration.vuejs.org/].
```
Comment (bojanrajh) [inline guides/updates/administration/vue3.md:12]: Please, update the link (revert () and []).

File: guides/updates/administration/vue3.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+Yes, there are tools and migration helpers that can automate certain aspects of the update process. You could start by enabling the Vue 3 rule set of `eslint`.
+
+### Where can I find support and community discussions about updating Shopware plugin to Vue 3?
+You can participate in discussions and seek help on the Shopware community Slack. There is a dedicated channel for this topic called `#vue3-update`.
+
+## External resources
+
+Here is a handpicked selection of external resources. This list provides a handy reference, granting you access to all the essential materials you might need.
+
+- (Vue 3 migration guide)[https://v3-migration.vuejs.org]
```
Comment (bojanrajh) [inline guides/updates/administration/vue3.md:46]: Update links syntax.

File: guides/updates/administration/vue3.md [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+```javascript
+Shopware.Application.view.vue3
+```
+
+### Check your templates
+
+For your templates to work correctly, follow the following in no specific order:
+
+- Replace all `sw-field` usages with the corresponding components.
+- (Check all v-models)[https://v3-migration.vuejs.org/breaking-changes/v-model.html]
```
Comment (bojanrajh) [inline guides/updates/administration/vue3.md:80]: Replace links syntax.

File: guides/updates/administration/vue3.md [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+### Using slots programmatically
+It is no longer sufficient to check if `this.$slots` has a property with the slot name to see if that slot exists. Instead, you must verify if your `slotName` contains an actual `v-node`.
+
+### this.$parent
+`this.$parent` is prone to errors because Vue 3 wraps the `AsyncWrapperComponent` around asynchronous components. Leading to the virtual dom tree to differ from Vue 2 to Vue 3. Where in Vue 2, a `this.$parent` call was successful, in Vue 3, a `this.$parent.$parent` may be necessary.
+
+### Vue dev tools performance
+Vue dev tools cause massive performance issues with huge Vue 3 applications.
+There is an open (issue on Github)[https://github.com/vuejs/devtools/issues/1875] with next to no activity from the maintainers.
```
Comment (bojanrajh) [inline guides/updates/administration/vue3.md:105]: Link syntax. :)

**Review by bojanrajh — CHANGES_REQUESTED** [review]

File: guides/updates/administration/vue3.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+App-based extensions aren't affected by these changes. However, if your extension is plugin-based and contains custom administration code, you likely need to do some refactoring.
+
+### Are there any breaking changes I should be aware of?
+Yes, Vue 3 introduced breaking changes. It's crucial to review the migration guide provided by Vue.js and this document for detailed information.
+
+### What steps should I follow to update my Shopware plugin to Vue 3?
+Typically, the process involves updating your project dependencies and modifying your code to adhere to Vue 3's API changes. Consult the Vue 3 documentation and this document's step-by-step instructions.
+
+### Can one plugin version be compatible with Shopware 6.5 and 6.6?
+Your plugin requires a new version in the Store. For instance, version `1.x` is for Shopware `6.5.x`, while version `2.0` is compatible with Shopware `6.6` and newer.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:28]: Maybe be a bit clear and say no here. So `No, your plugin requires`

File: guides/updates/administration/vue3.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+
+- (Vue 3 migration guide)[https://v3-migration.vuejs.org]
+- (Vue 3 breaking changes)[https://v3-migration.vuejs.org/breaking-changes/]
+- (Vue router migration guide)[https://router.vuejs.org/guide/migration/]
+- (Vue test utils migration guide)[https://test-utils.vuejs.org/migration/]
+
+## Step-by-step guide
+::: warning
+Depending on when you read this document, you still need to enable the `VUE3` feature flag.
+To be safe, enable the `VUE3` feature flag and restart your watcher.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:54]: Maybe add a link to the feature flag documentation?

File: guides/updates/administration/vue3.md [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+To follow along, you should have the following:
+
+- the latest Shopware `trunk` or an official release candidate
+- installed and activated your plugin
+- a running administration watcher (`composer run watch:admin`)
+
+### Update your plugin npm dependencies
+
+Make sure to align your `package.json` dependencies with the administration.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:65]: Maybe add a link to the admin `package.json`

File: guides/updates/administration/vue3.md [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+
+If you evaluate the following expression in the browser dev tools, the value `true` must be returned.
+
+```javascript
+Shopware.Application.view.vue3
+```
+
+### Check your templates
+
+For your templates to work correctly, follow the following in no specific order:
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:77]:
```suggestion
For your templates to work correctly, perform the following in no specific order:
```

File: guides/updates/administration/vue3.md [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+
+- Replace all `sw-field` usages with the corresponding components.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:79]: Link to the mapping file would be useful, `src/Administration/Resources/app/administration/src/app/component/form/sw-field/index.js` I guess

File: guides/updates/administration/vue3.md [hunk 10]
```diff
@@ -0,0 +1,129 @@
...
+
+Most of your code should be unaffected by the update. You can start by searching for `this.$`. The usage of `this.$` is an indicator of Vue's internal API. These calls are very likely to break except for `this.$tc`.
+
+If you have a lot of Vue internal API calls, check out the (Known issues section)[#known-issues].
+The best way to find errors is to test your application thoroughly, either by hand or automated.
+
+## Known issues
+
+### Lifecycle hooks
+Lifecycle hooks such as `@hook:mounted` may be triggered multiple times if the component is loaded asynchronously. Vue 3 will emit the hook for the `AsyncComponentWrapper` and the underlying component. Expect lifecycle hooks to be triggered more than once.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:95]: Is there an example in the admin so you can see how you might handle this if you really want something to run just once on mounted?

File: guides/updates/administration/vue3.md [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+
+### Using slots programmatically
+It is no longer sufficient to check if `this.$slots` has a property with the slot name to see if that slot exists. Instead, you must verify if your `slotName` contains an actual `v-node`.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:98]: I'm sure I used `$slots.mySlot` in a vue3 template the other day to check if a slot was provided, is that something different?

File: guides/updates/administration/vue3.md [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+
+### this.$parent
+`this.$parent` is prone to errors because Vue 3 wraps the `AsyncWrapperComponent` around asynchronous components. Leading to the virtual dom tree to differ from Vue 2 to Vue 3. Where in Vue 2, a `this.$parent` call was successful, in Vue 3, a `this.$parent.$parent` may be necessary.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:101]: Should we make a note to maybe prefer a different method of inter component communication?

File: guides/updates/administration/vue3.md [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+
+### Vue dev tools performance
+Vue dev tools cause massive performance issues with huge Vue 3 applications.
```
Comment (AydinHassan) [inline guides/updates/administration/vue3.md:104]:
```suggestion
Vue dev tools causes massive performance issues with huge Vue 3 applications.
```

File: guides/updates/administration/vue3.md (hunk 6 above)
Comment (seggewiss) [reply guides/updates/administration/vue3.md:54]: None of the feature flag documentation explains how to activate them just how to add one :see_no_evil:

File: guides/updates/administration/vue3.md (hunk 10 above)
Comment (seggewiss) [reply guides/updates/administration/vue3.md:95]: Maybe this is not clear enough :thinking: You cant make a hook in this combination only execute once. I will rewrite it to make it more clear.

File: guides/updates/administration/vue3.md (hunk 11 above)
Comment (seggewiss) [reply guides/updates/administration/vue3.md:98]: In Vue 3 $slots.MySlot should always be a render function. Accroding to the official docs you have to call this render function and see if it returns a v-node.

Comment (bojanrajh) [reply guides/updates/administration/vue3.md:98]: Small addition: check for the slot content is quite complex, depending on what you define as "empty" slot:
Discussion: https://github.com/vuejs/core/issues/4733#issuecomment-1244011992
WIP implementation (in vitepress): https://github.com/vuejs/vitepress/pull/2297/files#diff-eed5fa858ff4a0d3291cfeaa80f5426ed1948b75f25f9fafc002efe82a91bc36

**Review by AydinHassan — APPROVED** [review]

**Review by bojanrajh — APPROVED** [review]

File: guides/updates/administration/vue3.md (hunk 11 above)
Comment (seggewiss) [reply guides/updates/administration/vue3.md:98]: Thank you for sharing :+1: Pretty interesting discussion :blue_heart:

**Pull request merged**